Apps/phtwitter/twitter_connector.py

```diff
@@ -0,0 +1,242 @@
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # create and auth twitter object
+        t = self._create_twitter_object()
+
+        self.save_progress("Connecting to endpoint")
+
+        # get the list of tweets from the users timeline
+        try:
+            t.statuses.home_timeline(count=1)
+        except Exception as e:
+            self.save_progress(str(e))
+            return action_result.set_status(phantom.APP_ERROR)
+
+        self.save_progress("Successfully connected and authenticated")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_run_query(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # get the string to query twitter for
+        query = param['query']
+
+        # auth and search twitter
+        t = self._create_twitter_object()
+
+        # get time for searching
+        time_limit = self._get_date_limit()
+
+        try:
+            search_results = t.search.tweets(q=query, until=time_limit)
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, str(e))
+
+        # store the full results returned
+        action_result.add_extra_data(search_results)
+
+        search_results = search_results['statuses']
+
+        # create an empty dictionary to store tweet information
+        twitter_results = []
+
+        # The Twitter api returns a lot of information across tweets and re-tweets.
+        # This condenses everything and formats some of the tweets
+        # If you need the full results they are stored as extra data
+        for i in search_results:
+            tweet_info = {}
+            # gets the actual tweet. remove new line charactes
+            tweet_info["tweet"] = i['text'].replace("\n", "")
+            tweet_info["username"] = i["user"]["screen_name"]
+            if "retweeted_status" in i:
+                tweet_info["retweeted status"] = i["retweeted_status"]["text"]
+            else:
+                tweet_info["retweeted status"] = ""
+            hashtags = []
+            for tags in i["entities"]["hashtags"]:
+                hashtags.append(tags["text"])
+            tweet_info["hashtags"] = hashtags
+
+            tweet_urls = []
+            for url in i["entities"]["urls"]:
+                tweet_urls.append(url["expanded_url"])
+            tweet_info["urls"] = tweet_urls
+            # create a link to the actual tweet
+            tweet_info["Tweet Link"] = "https://twitter.com/%s/status/%s" % (i["user"]["screen_name"], str(i["id"]))
+            twitter_results.append(tweet_info)
+
+        action_result.add_data(twitter_results)
```

Comment:
I'd keep this in the for loop.

Reply:
As discussed on the slack, we are keeping it as is.